pkg/webhook/internal/cert/writer/secret.go

```diff
@@ -127,39 +112,30 @@ func (s *secretCertWriter) overwrite() (
 	if err != nil {
 		return nil, err
 	}
-	if s.dryrun {
-		return certs, s.dryrunWrite(secret)
-	}
 	err = s.Client.Update(nil, secret)
 	return certs, err
 }
 
-func (s *secretCertWriter) dryrunWrite(secret *corev1.Secret) error {
-	sec, err := yaml.Marshal(secret)
-	if err != nil {
-		return err
-	}
-	_, err = s.Writer.Write(sec)
-	return err
-}
-
 func (s *secretCertWriter) read() (*generator.Artifacts, error) {
-	if s.dryrun {
-		return nil, notFoundError{}
-	}
 	secret := &corev1.Secret{}
 	err := s.Client.Get(nil, *s.Secret, secret)
 	if apierrors.IsNotFound(err) {
 		return nil, notFoundError{err}
 	}
-	return secretToCerts(secret), err
+	certs := secretToCerts(secret)
+	if certs != nil {
+		// Store the CA for next usage.
+		s.CertGenerator.SetCA(certs.CAKey, certs.CACert)
```

Comment:
I am assuming secret will be updated only if installer is enabled ?

Reply:
Not always true.
Disabling the installer only skip installation during bootstrapping.
The webhook server will refresh its cert if the the cert is expiring.
IMO installer only cover bootstrapping time, after that the server may try to rotate its cert if necessary. If it doesn't have permission it will fail. 
We probably want to reconsider if it should fail or just log a warning.